sway-lsp/src/traverse/typed_tree.rs

```diff
@@ -1069,88 +1066,6 @@ impl Parse for ty::TyReassignment {
     }
 }
 
-impl Parse for ty::TyStorageReassignment {
-    fn parse(&self, ctx: &ParseContext) {
-        // collect storage keyword
-        if let Some(mut token) = ctx
-            .tokens
-            .try_get_mut(&to_ident_key(&Ident::new(
-                self.storage_keyword_span.clone(),
-            )))
-            .try_unwrap()
-        {
-            token.typed = Some(TypedAstToken::TyStorageResassignment(Box::new(
-                self.clone(),
-            )));
-            if let Some(storage) = ctx.namespace.get_declared_storage(ctx.engines.de()) {
-                token.type_def = Some(TypeDefinition::Ident(storage.storage_keyword));
-            }
-        }
-        if let Some((head_field, tail_fields)) = self.fields.split_first() {
-            // collect the first ident as a field of the storage definition
-            if let Some(mut token) = ctx
-                .tokens
-                .try_get_mut(&to_ident_key(&head_field.name))
-                .try_unwrap()
-            {
-                token.typed = Some(TypedAstToken::TypeCheckedStorageReassignDescriptor(
-                    head_field.clone(),
-                ));
-                if let Some(storage_field) = ctx
-                    .namespace
-                    .get_declared_storage(ctx.engines.de())
-                    .and_then(|storage| {
-                        // find the corresponding field in the storage declaration
-                        storage
-                            .fields
-                            .into_iter()
-                            .find(|f| f.name.as_str() == head_field.name.as_str())
-                    })
-                {
-                    token.type_def = Some(TypeDefinition::Ident(storage_field.name));
-                }
-            }
-            // collect the rest of the idents as fields of their respective types
-            for (field, container_type_id) in tail_fields
```

Comment:
Yeah it looks like it's this block that had the go to logic working. Inspecting the AST and it seems we can move this logic when looping over the fields in `TyExpressionVariant::StorageAccess`

```rust
TyStorageAccess {
    fields: [
        TyStorageAccessDescriptor {
            name: BaseIdent {
                name_override_opt: None,
                span: Span {
                    src (ptr): 0x000000010ef27890,
                    path: Some(
                        "/private/var/folders/3z/79mvj_fn5298rz6zb4gnctw00000gn/T/SWAY_LSP_TEMP_DIRQs3zQJ/storage/src/main.sw",
                    ),
                    start: 345,
                    end: 349,
                    as_str(): "var1",
                },
                is_raw_ident: false,
            },
            type_id: TypeId(
                67840,
            ),
            span: Span {
                src (ptr): 0x000000010ef27890,
                path: Some(
                    "/private/var/folders/3z/79mvj_fn5298rz6zb4gnctw00000gn/T/SWAY_LSP_TEMP_DIRQs3zQJ/storage/src/main.sw",
                ),
                start: 345,
                end: 349,
                as_str(): "var1",
            },
        },
        ..
    ]
}
```